components/region_cache_memory_engine/src/range_manager.rs

```diff
@@ -0,0 +1,211 @@
+// Copyright 2024 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::collections::{BTreeMap, BTreeSet};
+
+use engine_traits::CacheRange;
+
+use crate::engine::SnapshotList;
+
+#[derive(Debug, Default)]
+pub struct RangeMeta {
+    range_snapshot_list: BTreeMap<CacheRange, SnapshotList>,
+    ranges_evcited: BTreeSet<CacheRange>,
```

Comment:
When can entries in `ranges_evcited` or `ranges_unreadable` be deleted? Otherwise, these two itself would consume too much memory